src/lib/components/molecules/option-picker/index.jsx

```diff
@@ -0,0 +1,84 @@
+import { useState, useCallback } from "preact/hooks"
+import { mergeStyles } from "$styles/helpers/mergeStyles"
+import defaultStyles from "./style.module.css"
+
+/** @typedef {  "vertical" | "horizontal" } OptionLayoutDirection */
+
+export function OptionPicker({
+  title,
+  options,
+  layoutDirection = "horizontal",
+  onSelect,
+  styles,
+}) {
+  styles = mergeStyles(defaultStyles, styles)
+
+  const [selectedIndex, setSelectedIndex] = useState(0)
+
+  const onOptionClick = useCallback(
+    (option, index) => {
+      if (index === selectedIndex) return
+
+      setSelectedIndex(index)
+      if (onSelect) onSelect(index, option)
+    },
+    [selectedIndex, onSelect],
+  )
+
+  return (
+    <div className={styles.optionPicker}>
+      {title && <span className={styles.title}>{title}</span>}
+      <div className={[styles.options, layoutDirection].join(" ")}>
+        {options.map((option, index) => {
+          const isSelected = index === selectedIndex
+          return (
+            <button
+              key={option.title}
+              className={[
+                styles.option,
+                isSelected ? styles.selected : "",
+              ].join(" ")}
+              onClick={() => onOptionClick(option, index)}
+            >
+              <div className={styles.optionIconContainer}>
+                {typeof option.icon === "string" ? (
+                  <img src={option.icon} className={styles.optionIcon} />
```

Comment:
Suggestion: if an alt tag is out the question, I'd add `role=presentation` here.